### PR TITLE
Generic "don't show this message again" modal

### PR DESCRIPTION
**Pull request by justinvdm**

### CODE REVIEW

**Comment by justinvdm** [comment]
Realised there is a small change a I need to make, not yet ready for review.

File: go/base/static/js/src/components/views.js [hunk 1]
```diff
@@ -81,9 +82,70 @@
...
+      if (options.optional) { this.optional = options.optional; }
+
+      this.dontShow = false;
+      this.content = options.content || '';
+    },
+
+    onOk: function() {
+      if (this.optional) {
+        this.dontShow = this.$('.dont-show').is(':checked');
+        console.log(this.dontShow);
```
Comment (hodgestar) [inline go/base/static/js/src/components/views.js]: Stray log?

File: go/base/static/js/test/tests/components/views.test.js [hunk 2]
```diff
@@ -81,4 +82,92 @@ describe("go.components.views", function() {
...
+
+      // Remove animation to make phantomjs happy
+      // ('shown' and 'hidden' events don't seem to be triggered otherwise)
+      confirm.$el.removeClass('fade in');
+    });
+
+    afterEach(function() {
+      $('.modal').remove();
+      confirm.remove();
+      confirm.$el.remove();
```
Comment (hodgestar) [inline go/base/static/js/test/tests/components/views.test.js]: Should `confirm.remove()` perhaps remove the element and the modal if they happen to be around?

**Comment by hodgestar** [comment]
Other than the stray log call and the `confirm.remove()` question. :+1:.

File: go/base/static/js/src/components/views.js (hunk 1 above)
Comment (justinvdm) [inline go/base/static/js/src/components/views.js]: Ah, thanks.

File: go/base/static/js/test/tests/components/views.test.js (hunk 2 above)
Comment (justinvdm) [inline go/base/static/js/test/tests/components/views.test.js]: Ya, I think we only need one of these three lines. Was left behind after trying to find a bug, will fix this.

**Comment by justinvdm** [comment]
Ready for re-review.

File: go/base/static/js/test/tests/components/views.test.js [hunk 3]
```diff
@@ -81,4 +82,84 @@ describe("go.components.views", function() {
     });
   });
 
+  describe.only(".ConfirmView", function() {
```
Comment (hodgestar) [inline go/base/static/js/test/tests/components/views.test.js]: Stray .only?

File: go/base/static/js/test/tests/components/views.test.js [hunk 4]
```diff
@@ -81,4 +82,84 @@ describe("go.components.views", function() {
...
+    describe(".render", function() {
+      it("should show the modal", function() {
+        assert(noElExists('.modal'));
+        confirm.render();
+        assert(oneElExists('.modal'));
+      });
+    });
+
+    describe(".show", function() {
+      it("should show the modal if the user has disabled the modal",
```
Comment (hodgestar) [inline go/base/static/js/test/tests/components/views.test.js]: Should the "if the user has disabled the modal" text be in this descriptiong? The test doesn't seem to disable the modal?

File: go/base/static/js/test/tests/components/views.test.js [hunk 5]
```diff
@@ -81,4 +82,84 @@ describe("go.components.views", function() {
...
+
+      it("should trigger an 'ok' event if the user has disabled the modal",
+      function(done) {
+        confirm.on('ok', function() { done(); });
+
+        confirm.render();
+        confirm.$('.dont-show').click();
+        confirm.$('.ok').click();
+      });
+    });
```
Comment (hodgestar) [inline go/base/static/js/test/tests/components/views.test.js:141]: I think we are now missing a test that checks that the modal isn't displayed after ".dont-show" has been set but that "ok" is still triggered?

File: go/base/static/js/test/tests/components/views.test.js (hunk 3 above)
Comment (justinvdm) [inline go/base/static/js/test/tests/components/views.test.js]: Ah, well caught, thanks. Sorry, not my greatest coding day :/

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**